.github/workflows/deploy-sdk.yml

```diff
@@ -125,10 +133,22 @@ jobs:
   deploy-sonatype:
     name: Deploy SDK to Maven Central
     needs: [deploy-git-tag]
```

Comment:
Thank you for the clarification Levi, I thought `deploy-git-tag` would fail hence leading up to `deploy-sonatype` not working. But seems like it triggers and skips and doesn't fail so it will not stop the deployment job from failing.

A couple of more things, in the case of manual workflow, if `inputs.existing-git-tag` is empty, it would result in using the ref branch as the source of checkout right? just want to make sure we have the latest version. in the manual script, I would fetch the latest tag in case none was mentioned while dispatching the workflow. 

Secondly, 
```
uses: ./.github/actions/setup-android
      - name: Push to Sonatype servers
        run: MODULE_VERSION=${{ needs.deploy-git-tag.outputs.new_release_version }} ./scripts/deploy-code.sh
```

This script is dependent on `needs.deploy-git-tag.outputs.new_release_version` but if the `deploy-git-tag` is skipped, would it populate this variable? i wouldn't believe so.

Reply:
Good catch. I did not consider that since the iOS SDK doesn't update the version at compile-time. 

Until we can test deployment CI workflows before merging PRs, I think this PR is not worth merging. Some copy/pasted code in the short-term is OK for now. I will close this PR to do the refactor another time.